ggshield/core/config/user_config.py

```diff
@@ -63,14 +64,81 @@ def validate_policy_id(policy_id: str) -> bool:
     return bool(POLICY_ID_PATTERN.fullmatch(policy_id))
 
 
-def validate_policy_ids(values: Iterable[str]) -> None:
-    invalid_excluded_policies = [
-        policy_id for policy_id in values if not validate_policy_id(policy_id)
-    ]
-    if len(invalid_excluded_policies) > 0:
-        raise ValidationError(
-            f"The policies {invalid_excluded_policies} do not match the pattern '{POLICY_ID_PATTERN.pattern}'"
-        )
+@marshmallow_dataclass.dataclass
+class ConfigIgnoredElement(FilteredConfig):
+    comment: Optional[str]
+    until: Optional[datetime]
+
+    @abstractmethod
+    def short_repr(self):
+        return "Element"
+
+    # Accept date yyyy-mm-dd instead of a full datetime
+    @pre_load
+    def parse_date(self, data, **kwargs):
+        if not (isinstance(data, dict)) or data.get("until") is None:
+            return data
+        try:
+            data["until"] = str(datetime.strptime(data["until"], "%Y-%m-%d"))
+            return data
+        except ValueError:
+            return data
+
+    @post_load
+    def datetime_to_utc(self, data, **kwargs):
+        if data["until"] is not None:
+            data["until"] = data["until"].astimezone(timezone.utc)
+        return data
+
+
+def validate_ignored_elements_array(data: dict, field: str):
+    valid_elements = []
+    for ignored in data[field]:
+        if ignored.until is None or ignored.until > datetime.now(tz=timezone.utc):
+            valid_elements.append(ignored)
+        else:
+            display_warning(
+                f"{ignored.short_repr()} has an expired 'until' "
+                f"date ({ignored.until}), please update your configuration file."
+            )
+    data[field] = valid_elements
+    return data
+
+
+@marshmallow_dataclass.dataclass
+class IaCConfigIgnoredPath(ConfigIgnoredElement):
+    path: str
+
+    # If we set default values in the parent class,
+    # we would need to set them for the child class attributes too
```

Comment:
I don't understand this comment. Which default values are we talking about here?

Reply:
I tried to set default values for attributes in the parent class:
```
class ConfigIgnoredElement(FilteredConfig):
    comment: Optional[str] = None
    until: Optional[datetime] = None
```
It resulted in an error for child classes with required attributes. Seemingly the parent attributes are constructed first:
`TypeError: non-default argument 'path' follows default argument`

Reply:
I see. Indeed, this is to be expected. Not sure we need this comment.